src/parser/xasl_generation.c

```diff
@@ -13960,6 +13966,27 @@ pt_to_outlist (PARSER_CONTEXT * parser, PT_NODE * node_list, SELUPD_LIST ** selu
 		      regu->value.dbvalptr = value_list->val;
 		      /* move to next db_value holder */
 		      value_list = value_list->next;
+		      if (node->info.query.q.select.hint & PT_HINT_NO_SUBQUERY_CACHE)
+			{
+			  /* it means SUBQUERY RESULT won't be cached. */
+			}
+		      else
+			{
+			  if (pt_check_corr_subquery_hash_result_cache (xasl))
+			    {
+			      bool cachable = true;
+			      parser_walk_tree (parser, node, NULL, NULL, pt_check_corr_subquery_not_cachable_expr,
```

Comment:
How about checking it in pt_check_corr_subquery_hash_result_cache()?

Reply:
Since pt_check_corr_subquery_hash_result_cache() is called recursively, it would be a good idea to create a new function.

Reply:
The `pt_check_corr_subquery_hash_result_cache` function recursively searches the created XASL structure, while `pt_check_corr_subquery_not_cachable_expr` recursively searches the created PT_NODE structure. Combining these two would likely complicate the implementation.

Reply:
I have understood the intent behind your review and applied it to the code. Please check and confirm.